agent/task_context.go

```diff
@@ -519,3 +520,20 @@ func (tc *taskContext) getUserEndTaskResponse() *triggerEndTaskResp {
 
 	return tc.userEndTaskResp
 }
+
+func (tc *taskContext) deviceNames(ctx context.Context) error {
+	if tc.taskConfig == nil || tc.taskConfig.Distro == nil {
+		return nil
+	}
+
+	partitions, err := disk.PartitionsWithContext(ctx, false)
```

Comment:
nit: consider returning early if `tc.taskConfig.Distro.Mountpoints` is empty to avoid asking for the partitions unnecessarily.

Reply:
done